### PR TITLE
[Semi-Modular] SeleneStation [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by sergeirocks100**
## About The Pull Request

s85

Serene is a medium to high-pop map, originally from Fulp. It was made by Helianthus, not myself, full credits for the base map from Fulp goes to them.

I think it's a pretty unique and interesting map, and i think it would fit in well on Skyrat's map rotation.

## How This Contributes To The Skyrat Roleplay Experience

Variety is the spice of life, and i think Selene will bring some variety to the map rotation.

Plus, it's not attached to a map on the /tg/ mainline, so it allows for more mapping freedom.

## Changelog
:cl:
add: Nanotrasen space station 621-A, codename "Selene", has been put into service.
/:cl:

### CODE REVIEW

**Comment by FlamingLily** [comment]
This map looks fucking awesome

No annoying gimmick

A robust layout that isn't just Box/Meta/Delta

Sign me the fuck up

File: _maps/_basemap.dm
Comment (FlamingLily) [inline _maps/_basemap.dm]: Skyrat edit comment

**Review by FlamingLily — CHANGES_REQUESTED** [review]
The mining shuttle's still referencing fulp, be sure to fix that

**Comment by softcerv** [comment]
Looks neat, although I do wonder why there are two viro workstations

**Comment by Trubbl3** [comment]
something new is always welcomed

**Comment by GoldenAlpharex** [comment]
Looks good! If I may suggest, I would put some reinforced windows here, to improve the visibility (and overall look) of Xenobiology:
image

Here, I would move the cyborg charger to this stall (in red) and then turn the cyborg charger spot into some showers (in green), because the only showers being kind of open in the locker room is a bit awkward:
image

This is cool, but are you sure that people won't hurt themselves big time going in and out of that room? Also, what's the point of barricading that part off?
image

This is a lot of free tools lockers:
image

If you wanted to be real cool, you could add some windoors here, to stop cross-polinisation, or rather, toggle it on/off depending on what the botanists would like. Could be a nice little QoL:
image

This feels like a bit too much protection for these boards:
image

There's also no Blueshield or NT Rep office.

Overall, it looks really nice and I can't wait to try it out.

**Comment by Gandalf2k15** [comment]
We can test this next round.

**Comment by GoldenAlpharex** [comment]
Whoops, I think those weren't the right bedsheets.
image

**Comment by MidoriWroth** [comment]
Hydroponics trays seem to consider water a nutrient, and the water in the trays rapidly fluctuates
S8qxapq

**Comment by axietheaxolotl** [comment]
> 
> 
> Whoops, I think those weren't the right bedsheets.
> image

This isn't the fault of the map creator. This is the fault of me being autistic. Double bedsheets now spawn in the random sheet obj. Which is being fixed soon

**Comment by GoldenAlpharex** [comment]
> > Whoops, I think those weren't the right bedsheets.
> > image
> 
> This isn't the fault of the map creator. This is the fault of me being autistic. Double bedsheets now spawn in the random sheet obj. Which is being fixed soon

🎊

**Comment by axietheaxolotl** [comment]
C+P from Discord:
`This map looks good, but it's missing a lot because if it's port status. People have already mentioned there's no NTRep or Blueshield office, but as well: there's no Command Clothing Vendor, there's no SecMed locker (it could also deal with a healthy sprinkling of our custom airlocks)`

**Comment by SpookyTheFox** [comment]
Looks really good! A few first impressions for possible improvements, I will edit to add to this list as I look closer at it:
I'm mostly focusing in mebay, because that's what I main. Overall, very very good!
A+ on the lobby/reception area. I love everything about it. Public facing chemistry counter, a reception desk, sofas, tables, lamps, potted plants... excellent!
The chemistry plumbing room- could use more interior windows facing into the hallway, preferably also with shutters like the ones on other maps. Ideally, the windows should expose the chem dispenser set to the hallway, so it can be upgraded by someone walking by with a BSRPED. It also needs a second door, probably just into maintenance in the south.
The recovery room- makes sense ICly, but usually anyone afk while in medbay gets just put out in the lobby. Consider possibly cutting it up into another exam room or something.  
Plasma man/vox medbay is really cool! It needs the air alarm set to not be in alarm state at round start though. Maybe some signage of some sort to indicate what it is. While the chutes do their job, doors with tiny fans under them might do the job better. They will let people go in and out in a less... silly way.
I don't see a MULE chute anywhere in medbay.

Lots of good maintenance features I've seen so far, but it will take a while to look at all of it.

Rec area has weight machines- finally! Half my characters are _ripped_ and it makes no sense that they wouldn't be found in a gym environment at least some of the time.
AI sat access in the rec area is a little odd, maybe make that a strong door with reinforced walls on the part facing into the public area.
Seems like there's plenty of holopads, that's good too, I use them a lot as AI.

**Comment by CandleJaxx** [comment]
toxins/ordinance mass driver isn't connected to the controller
image
door is, driver not

**Comment by Sonic121x** [comment]
1:Engineering lacking a portable generator.(there is plasma but no portable generator in engineering,maybe consider replace one of the budget solar panel crate in power storage or secure storage)
2:Table in Kitchen(The one in the red box should be a rTable like the two in yellow box)
I2
3:Maintance door in HFR room
image

**Comment by JohnFulpWillard** [comment]
This has been really hurtful for me to watch unfold from the very start.

You tried porting this map to your own server, one that one of our mappers spent months making, but couldn't even bother to be respectful enough to properly spell (calling it 'Serene' instead of 'Selene') until the creator of the map themselves (Helianthus) commented on your original PR about how hurt they are that you are deliberately misspelling their map's name and that you couldn't even bother to ask them beforehand, to which you didn't respond to at all (but at least you are using the right name for the map in the TITLE, but you still misspelled it in the PR's body, so you've obviously seen the comment they left).

Then, a few days after their comment, you are now porting it to ANOTHER server, and AGAIN without asking the original map maker first (something that takes a minute to ask over discord, there's even a dedicated channel for this sort of stuff), nor did you again bother crediting the original maker of the map (A maintainer had to edit it in for you instead) https://github.com/fulpstation/fulpstation/pull/109

I don't know what Fulpstation did to you or why you are doing this, whether it's out of spite or grudge against the server or the map maker as an individual, but what you've done and have been doing has been really disrespectful and personally comes off to me as doing it maliciously or out of hate.

And this isnt even to mention how the entire time you've been working on porting this map, you couldn't even say anything nice about it yourself from your own public server.
image

**Comment by Helianthus00** [comment]
Please be aware that I absolutely hate the idea of my map being used on an ERP server, though I have no way to stop you.

Though I'm not sure why you're porting something to this "shitty codebase" as you called it. It is disgusting, your lack of respect for fellow contributors in the community.

**Comment by hugmeorelse** [comment]
> Please be aware that I absolutely hate the idea of my map being used on an ERP server, though I have no way to stop you.
> 
> Though I'm not sure why you're porting something to this "shitty codebase" as you called it. It is disgusting, your lack of respect for fellow contributors in the community.

I don't know what you have against ERP servers (nor why you would specify that as if we're tainted), but I have to agree that it seems pretty bad faith on the part of the person here to port the map out of spite. We have no plans of merging this any time soon out of respect. Let me confer with my maintainers and get their take as a formality, but we will most likely close this PR.

**Comment by tf-4** [comment]
-1
Lovely map, ported in the most disrespectful, unbecoming way possible.
Not at all interested in seeing it on Skyrat for that reason alone - which is somewhat of a shame, as it is legitimately a nice map.

**Comment by FlamingLily** [comment]
As much as I hate to say it, I have to agree with the above.

This map is a really wonderful map, it's balanced, well designed, it's just fantastic all around, from what I've seen.

But we shouldn't merge it without the creator's wishes.

**Comment by Helianthus00** [comment]
> I don't know what you have against ERP servers (nor why you would specify that as if we're tainted), but I have to agree that it seems pretty bad faith on the part of the person here to port the map out of spite. We have no plans of merging this any time soon out of respect. Let me confer with my maintainers and get their take as a formality, but we will most likely close this PR.

I am also a Fulpstation admin, as well as mapper. I have nothing personally against ERP servers, but I do have administrative issues with them. I won't go into it here, but I can say I don't hate Skyrat or its community (and especially not the contributors, as prolific and creative as they are) so please don't take that with offense. 

My real issue is the person PRing this and the way they went about it. The whole things with the map more has to do with alterations I find distasteful, which would certainly be necessary for a server like this.

**Comment by GoldenAlpharex** [comment]
-1
We can port it ourselves at a later date, without doing so in a disrespectful way. I don't support ripping stuff out of somewhere else without even asking first, nor properly crediting them.

Also, porting something out of spite to another codebase while also being extremely condescending about it is never something I could ever support.

A shame, really.

**Comment by tralezab** [comment]
sergei is really toxic in the tg community as well i suggest sending this spiteful man packing

**Comment by KathyRyals** [comment]
EXTREMELY good map, superb asthetics, I am genuinely interested in porting it to /tg/ if anyone would be so kind as to give me Helianthus00 discord handle to talk about it first. I'm Cimika on the tg cord.

Edit : Also obligatory "Sergei Moment"

**Comment by LemonInTheDark** [comment]
Good map, needs more airlocks.

Stop being a cock serge It'll get you nowhere at breakneck speed

**Comment by sergeirocks100** [comment]
I may as well try to address the shit stew that i boiled for myself.

First, lets talk about Fulp's allegations and concerns.

Selene/Serene is not really unique in terms of me spelling things wrong on purpose, i actually do it with a few other things, such as spelling GAGS as "GAS", and spelling Ervin Bransfield's name as "Erwin".

It actually originated as a honest misremembering of the map's name, until it entered my lexicon of things i say wrong on purpose.

However, if it truly does bother the people responsible for the map in question, i will start calling it by its proper name.

I had no clue at all that the map's creator had replied to the PR, as i stopped looking at it after i merged it.

Being that the map was CC-BY-SA, i didn't think i would have to ask for permission to use the map, or even that the author would have cared that i used the map without asking him.

Had i known that the creator would have been displeased had i not, i would have asked for permission first.

Anyways, now lets get on to the story of how i ended up porting the map to Skyrat in the first place.

From pretty much the moment i had managed to open the map on StrongDMM, i had noticed quite a few design decisions that i didn't think were very good, such as the permabrig and brig being halfway across the station from eachother, which caused me to have a low opinion of the map.

I also was not a fan of Skyrat as a codebase very much, mostly because it had a lot of things that i disliked on /tg/, which Skyrat is a direct downstream of, and some other things i disliked on Skyrat's side.

Therefore, i decided i would try to port Selene to Skyrat, and see how what i thought was a bad map fared on Skyrat.

I saw a map like Outpost get merged successfully on Skyrat, so i figured that Selene wouldn't fare all that badly here.

So i went ahead, spent about a hour of initial work or so to get the map to Skyrat mapping standards, and then i made the PR.

As Fulp said, i had forgotten to credit the map's creator properly on the PR, but that was a mistake on my end, and Golden fixed it for me.

To my utter surprise, the map was not only tolerated, but praised.

It had even gotten test merged just 7 hours after i made the PR.

I then spent the next day or two in a state of disbelief that the map had actually fared well, and i spent a little while fixing bugs on the map that people had alerted to me, all the while, talking about, in probably harsher words than i should have, the goings on with the map on the Manuel discord.

Then Fulp brought all of my fuckups into the public light, and you know the story from there.

For all that has happened, i do think that i owe some apologies.

I apologize to Helianthus, for not asking for his permission, not properly crediting him, misspelling the name of his map on purpose, and insulting the map.

I apologize to Skyrat as a community, not only for insulting the codebase, but for trying to port a map that i thought they would dislike to the codebase, even if they actually liked it, and provided helpful feedback for how to improve it.

And finally, i think i should apologize to Fulpstation, for the stuff i did to Helianthus, a admin and contributor for them.

I won't be closing this PR, as people actually liked the map, however troubled its past may be, and i want to leave the map's fate in the hands of the Skyrat development staff.

**Comment by KasparoVy** [comment]
The end of this comment section feels net-neutral
The map is awesome and hopefully all the contribution/respect/courtesy issues can be resolved. Np if it doesn't get in, the work of the OG authors is greatly appreciated either way

Had no idea this was a Fulpstation map until now, seems like one many folks would enjoy

**Comment by Gandalf2k15** [comment]
Given that we have too many maps as it is, we will not be adding this one.

**Pull request closed without merging**
